src/main/java/com/autotune/database/service/ExperimentDBService.java

```diff
@@ -142,10 +141,63 @@ public void loadAllPerformanceProfiles(Map<String, PerformanceProfile> performan
+         * Incorporate a buffer period of "threshold days" to account for potential remote cluster downtime.
+         * This adjustment aims to align the cumulative hours' duration with LONG_TERM_DURATION_DAYS.
+         * Additionally, there's an added layer of coverage involving the utilization of a LIMIT.
+         * This measure ensures that the total duration in hours corresponds precisely to LONG_TERM_DURATION_DAYS.
+         */
+        cal.add(Calendar.DAY_OF_MONTH, -(KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS +
+                KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS_THRESHOLD));
+        // Get the new Timestamp after subtracting 10 days
+        fromTime = new Timestamp(cal.getTimeInMillis());
+        /*
+            To restrict the number of rows in the result set, the Load results operation involves locating the appropriate method and configuring the desired limitation.
+            It's important to note that in order for the Limit rows feature to function correctly,
+            the CreateExperiment API must adhere strictly to the trail settings' measurement duration and should not allow arbitrary values
+        */
+        KruizeObject kruizeObject = mainKruizeExperimentMap.get(experimentName);
+        int limitRows = (int) ((
+                KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS *
+                        KruizeConstants.DateFormats.MINUTES_FOR_DAY)
+                / kruizeObject.getTrial_settings().getMeasurement_durationMinutes_inDouble());
+        LOGGER.debug("Limit rows set to {}", limitRows);
         ExperimentInterface experimentInterface = new ExperimentInterfaceImpl();
         // Load results from the DB and save to local
-        List<KruizeResultsEntry> kruizeResultsEntries = experimentDAO.loadResultsByExperimentName(experimentName, interval_end_time, limitRows);
+        List<KruizeResultsEntry> kruizeResultsEntries = experimentDAO.loadResultsByExperimentName(experimentName, fromTime, toTime, limitRows);
+        /*
+            Determine whether the sum of durations matches the value of the long-term duration.
+            If they do not match, execute a query that selects all rows with interval_end_times less than a specified threshold.
+            The selection should be limited based on the formula: (LongTermDuration_days * 60 mins) / trailsettings.duration_in_mins.
+         */
+        double sum_of_duration_in_mins = 0.0;
+        for (KruizeResultsEntry entry : kruizeResultsEntries) {
+            sum_of_duration_in_mins += entry.getDuration_minutes();
+        }
+
+        if (sum_of_duration_in_mins < KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS * KruizeConstants.DateFormats.MINUTES_FOR_DAY) {
```

Comment:
What happens when we have more records than is needed. Since you are setting the threshold as 10 days, you can get significantly more records than needed.